xds/src/main/java/io/grpc/xds/internal/rbac/engine/GrpcAuthorizationEngine.java

```diff
@@ -335,15 +335,17 @@ private String getPath() {
 
     @Nullable
     private String getHeader(String headerName) {
-      if ("TE".equals(headerName)) {
+      headerName = headerName.toLowerCase();
```

Comment:
`toLowerCase(Locale.ROOT)` (or equivalently for this usage, `Locale.US`). We rarely want to use the default locale.